### ISSUE TITLE
Cases > Filter by "date of outcome" [1]

### ISSUE THREAD

**Issue by Candice-Louw** [hzi-braunschweig/SORMAS-Project#2920]
### Situation Description
It is currently not possible to filter Cases based on the "outcome date" of the Case.  This statistic is used for daily reporting by health departments (GAs and the CH Cantone confirmed) to establish how many Cases passed away and recovered on a specific date  (currently possible to filter only if "outcome date" is the same as the "report date" - see: https://github.com/hzi-braunschweig/SORMAS-Switzerland/issues/34#issuecomment-692564000 and https://github.com/hzi-braunschweig/SORMAS-Project/issues/2832).

### Feature Description
(Step 1) Cases > Case Directory > Filter: Outcome of case [No Outcome Yet/Deceased/Recovered/Unknown] - select "Deceased" or "Recovered" (only once one of these are selected should the "date of outcome" become filterable) > (Step 2) Show more filters > Filter: Case reference date [Most relevant date/Symptom onset date/Case report date] - add "Date of Outcome" here if "Outcome of case" filter (in step 1) is set to "Deceased" or "Recovered".

This "Date of Outcome" is taken from Cases > Open a Case (Case information) > Outcome of case ("Deceased" or "Recovered") > Date of Outcome (conditional visibility).

### Additional Information
![image](https://user-images.githubusercontent.com/65529128/93736286-25952980-fbe0-11ea-9b18-8f1f18356e67.png)

Please be sure to update tables/other filtering related items in the event of this date being empty that no error will occur (e.g. when filtering tables by clicking on the heading).

**Comment by bernardsilenou** [comment]
@Candice-Louw Can we add this in the statistics section instead? Or if they need to know daily death, then we can add to dashboard.
The reason is that we first need to filter by date of report and get the base line of cases, say N,
then out of this N, we then apply the second filter bases on date of outcome to get n
- n only makes sense based on N, because saying 10 people die in a day is not the same as saying 10/100000 die in a day.

**Comment by bernardsilenou** [comment]
@Candice-Louw I just had a second thought and realized that even showing just the daily death with out giving the domominator (total cases) is wrong. Politicians and journalist are often interested in number of daily indicators like death and falsely interpret them. So my recommendation is to  do exactly as you proposed but add it to statistics

**Comment by Candice-Louw** [comment]
@bernardsilenou - Statistics section should solve the problem, so yes, please go ahead with this solution.

**Comment by bernardsilenou** [comment]
@Candice-Louw I had a look at the statistics and realized what is lacking is the option to generate tables on a single date or date range, in addition to visualize tables by date of outcome, Here is a suggestion for you to have a look also if it can help
![image](https://user-images.githubusercontent.com/12996724/95927055-1314a700-0dbe-11eb-9aeb-1aac4bbe570f.png)

**Comment by markusmann-vg** [comment]
- [ ] add date of outcome as another filter option to the statistics

**Comment by StefanKock** [comment]
Some changes to do by @ChristopherRiedel coming from review.

**Comment by ChristopherRiedel** [comment]
Merging this resultet in a failing test (testQueryCaseCount in CaseStatisticsFacadeEjbTest):
`Caused by: java.lang.NullPointerException
	at java.util.Calendar.setTime(Calendar.java:1770)
	at de.symeda.sormas.api.utils.DateHelper.getEpiWeek(DateHelper.java:526)
	at de.symeda.sormas.backend.H2Function.epi_week(H2Function.java:24)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.h2.engine.FunctionAlias$JavaMethod.getValue(FunctionAlias.java:441)
	... 69 more`

**Comment by bernardsilenou** [comment]
@MateStrysewske @ChristopherRiedel  If you have a look at the figure, I specified that "add option for single gate and date interval"
I remember the users wanted to know the number of death per day

**Comment by MateStrysewske** [comment]
@bernardsilenou That's a topic for a separate issue though. This issue specifically is about adding date of outcome as an additional filter option.
